docs/src/account/id.md

```diff
@@ -0,0 +1,67 @@
+# Account ID
+
+> [!Note]
+> An immutable and unique identifier for the `Account`.
+
+The `Account` ID is a 120-bit long number. This identifier is designed to contain the metadata of an account. The metadata includes the [account type](#account-type), [account storage mode](#account-storage-mode) and the version of the `Account`. This metadata is included in the ID to ensure it can be read without needing the full account state.
+
+The ID is generated by hashing a randomly generated seed together with commitments to the initial code and storage of the `Account` until the resulting ID has the desired account type and storage mode. This process requires a small amount of Proof-of-Work (9 bits) that can be done even by low-powered devices. The resulting 256-bit hash is shortened to 120 bits.
+
+Account type and storage mode are chosen at account creation time and cannot be changed later.
+
+### Account type
+
+There are two main categories of accounts in Miden: **basic accounts** and **faucets**.
+
+- **Basic Accounts:**
+  Basic Accounts may be either mutable or immutable:
+  - *Mutable:* Code can be changed after deployment.
+  - *Immutable:* Code cannot be changed once deployed.
+
+- **Faucets:**
+  Faucets are always immutable and can be specialized by the type of assets they issue:
+  - *Fungible Faucet:* Can issue fungible [assets](../asset.md).
+  - *Non-fungible Faucet:* Can issue non-fungible [assets](../asset.md).
+
+### Account storage mode
+
+Users can choose whether their accounts are stored publicly or privately. The preference is encoded in the third and forth most significant bits of the account's [ID](#id):
+
+- **Public Accounts:**
+  The account's state is stored on-chain, similar to how accounts are stored in public blockchains like Ethereum.
+
+- **Network `Account`s:**
+  The account's state is stored on-chain, just like **public** accounts. Additionally, the network will monitor this account for any public notes targeted at it and attempt to create network transactions against the account, which consume the notes. Contracts that rely on a shared, publicly accessible state (e.g., a DEX) should be network accounts.
```

Comment:
```suggestion
  The account's state is stored on-chain, just like **public** accounts. Additionally, the network will monitor this account for any public notes targeted at it and attempt to create network transactions against the account, which consume the notes. Contracts that rely on a shared, publicly accessible state (e.g., an AMM) should be network accounts.
```

A DEX can be built without network accounts (e.g. a decentralized orderbook is a DEX), while AMMs (subset of DEXes) rely on shared public state.